File: src/main/kotlin/unizar/labis/g03/backendapp/services/CalculoOcupantesMaximos.java
package unizar.labis.g03.backendapp.services;

import unizar.labis.g03.backendapp.model.entities.Espacio;

import java.util.List;

public class CalculoOcupantesMaximos {

    public Integer  maximosOcupantesValido( List<Espacio> espacios){
        int suma = 0;
        for (Espacio espacio : espacios) {
            suma += espacio.getCapacidadMaxima();
        }
        return suma;
    }
}

File: src/main/kotlin/unizar/labis/g03/backendapp/domain/model/valueObjects/CalculoOcupantesMaximos.kt
package unizar.labis.g03.backendapp.domain.model.valueObjects

import org.springframework.stereotype.Service
import unizar.labis.g03.backendapp.domain.model.entities.Espacio

class CalculoOcupantesMaximos {
    fun calculo(espacios: List<Espacio>): Int {
        var suma = 0
        for (espacio in espacios) {
            suma += espacio.getCapacidadMaxima()
        }
        return suma
    }
}
